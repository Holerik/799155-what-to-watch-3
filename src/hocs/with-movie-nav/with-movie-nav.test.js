// with-movie-nav.test.js
import React from "react";
import renderer from "react-test-renderer";
import PropTypes from "prop-types";
import withMovieNav from "./with-movie-nav.jsx";

const MockComponent = (props) => {
  const {children} = props;

  return (
    <div>
      {children}
    </div>
  );
};

MockComponent.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired,
};

const MockComponentWrapped = withMovieNav(MockComponent);

it(`withMovieNav is rendered correctly`, () => {
  const tree = renderer.create(
      <MockComponentWrapped
        activeItem={1}>
      </MockComponentWrapped>).toJSON();

  expect(tree).toMatchSnapshot();
});
