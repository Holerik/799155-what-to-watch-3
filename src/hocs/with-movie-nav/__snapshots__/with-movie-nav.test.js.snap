// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withMovieNav is rendered correctly 1`] = `
<div>
  <nav
    className="movie-nav movie-card__nav"
  >
    <ul
      className="movie-nav__list"
      onClick={[Function]}
      onMouseOver={[Function]}
    >
      <li
        className="movie-nav__item
                  "
      >
        <a
          className="movie-nav__link"
          href="/moviecard-full"
        >
          Overview
        </a>
      </li>
      <li
        className="movie-nav__item
                  movie-nav__item--active"
      >
        <a
          className="movie-nav__link"
          href="#"
        >
          Details
        </a>
      </li>
      <li
        className="movie-nav__item
                  "
      >
        <a
          className="movie-nav__link"
          href="/moviecard-reviews"
        >
          Reviews
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
